apis/kueue/v1beta1/provisioningrequestconfig_types.go

```diff
@@ -57,6 +58,58 @@ type ProvisioningRequestConfigSpec struct {
 	// +listType=set
 	// +kubebuilder:validation:MaxItems=100
 	ManagedResources []corev1.ResourceName `json:"managedResources,omitempty"`
+
+	// retryStrategy defines strategy for retrying ProvisioningRequest.
+	// If null, then the default configuration is applied with the following parameter values:
+	// BackoffLimitCount:  3
+	// BackoffBaseSeconds: 60 - 1 min
+	// BackoffMaxSeconds:  1800 - 30 mins
+	//
+	// To switch off retry mechanism
+	// set retryStrategy.backoffLimitCount to 0.
+	//
+	// +optional
+	RetryStrategy *ProvisioningRequestRetryStrategy `json:"retryStrategy,omitempty"`
+}
+
+var (
+	DefaultRetryStrategy = ProvisioningRequestRetryStrategy{
+		BackoffLimitCount:  ptr.To(int32(3)),
+		BackoffBaseSeconds: 60,   // 1 min
+		BackoffMaxSeconds:  1800, // 30 min
+	}
+)
+
+type ProvisioningRequestRetryStrategy struct {
```

Comment:
Historically, timestamp was introduced to the WaitForPodsReady configuration before backoff configuration. It was the first way of putting some overhead on workloads that have failed. Since we already do it by doing backoff, this seemed redundant

Reply:
> Historically, timestamp was introduced to the WaitForPodsReady configuration before backoff configuration.

IIUC, this is slightly incorrect. Actually, both (timestamp and backoff) are discussed in the same KEP (https://github.com/kubernetes-sigs/kueue/pull/1311#issuecomment-1870531528), then the backoff was implemented a month later. So, both are needed for this enhancement.

> Since we already do it by doing backoff, this seemed redundant

As you can see the above, the timestamp and backoff is not same objective. The backoff is a mitigation solution for the timestamp. So, I still think both are needed.

Reply:
Discussed offline, we'll introduce `timestamp` field as a followup. For now Kueue relies on eviction timestamp in case of eviction by an admission check

Reply:
I have some discussions with @PBundyra offline.
In conclusion, we decided to use the Eviction time to requeue the Workload the same as the WaitForPodsReady so that we can mitigate the risk that the following Workloads are blocked by the evicted Workloads.

For sure, we can mitigate the risk to set longer backoff time to retryStrategy. But, this is needed to tune the backoff parameters, then the tuning make the usability worse.

OTOH, if we want to provide the know to set the timestamp field, we need to design / implement the mechanism to specify the queuing timestamp every Workloads. But the effort is not easy.

So, we postpone introducing the timestamp field to after the next release.

Reply:
I also discussed this with Patryk and the implementation of `timestamp: Creation` seems rather complex, and the need for it is unclear. So, I would be in favor of keeping the implementation as simple as possible and to defer the new field until we have such feedback from users - either in this KEP (before graduation), or follow up KEP.

Reply:
WDYT @tenzen-y ?

Reply:
Yes, that is key point for the timestamp field. 

Additionally, we need to design the timestamp field to mitigate quota violations by users since we do not want to allow batch users to turn timestamp to creation so that they can prioritize arbitrary workloads.

So, I would propose handling this timestamp feature as a separate comprehensive feature, not only for ProvisioningRequest. I did not say that the timestamp should be implemented.